cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -71,6 +81,184 @@ public Schema getSchema() {
     return (T) fields.get(fieldName);
   }
 
+  /**
+   * Get the {@link LocalDate} from field. The field must have {@link LogicalType#DATE} as its logical type.
+   *
+   * @param fieldName date field to get.
+   * @return value of the field as a {@link LocalDate}
+   * @throws UnexpectedFormatException if the field name is not present in schema or the provided field
```

Comment:
Why it is unexpected format if the field is missing in the schema? Also, this creates an inconsistent behavior with the simple `get` method, which doesn't check for field presence in the schema, but rather just return `null`. (BTW, the `get` method is missing the `@Nullable` annotation).

Reply:
`UnexpectedFormatException` is thrown from setters if the field is missing from schema. But to make it consistent with `get` method, we should return null.